cachi2/core/models/sbom.py

```diff
@@ -93,3 +93,133 @@ class Sbom(pydantic.BaseModel):
     def _unique_components(cls, components: list[Component]) -> list[Component]:
         """Sort and de-duplicate components."""
         return unique_sorted(components, by=lambda component: component.key())
+
+    def to_spdx(self) -> "SPDXSbom":
+        """Convert a CycloneDX SBOM to an SPDX SBOM."""
+        return SPDXSbom(
+            packages=[
+                SPDXPackage(
+                    name=component.name,
+                    version=component.version,
+                    package_download_location=component.purl,
+                )
+                for component in self.components
+            ],
+            creationInfo=SPDXCreationInfo(
+                creators=[f"{tool.vendor} {tool.name}" for tool in self.metadata.tools]
+            ),
+        )
+
+
+class SPDXPackageExternalRef(pydantic.BaseModel):
+    """SPDX Package External Reference.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: str
+    referenceLocator: str
```

Comment:
I would suggest using snake-case for fields: `reference_category` etc. Also `str` is a very wide type, are there any limitations on what could be a reference locator, category and type?

Reply:
Yeah, I can add some validation there if you would like to have more strict.

Reply:
also for snake case style, I'm also rather fan of snake-case style but according to standard spdx use camel case, so that's why.

Reply:
My point is that whatever `referenceLocator` is, it is more than just a `str`. Symbols within this comment field form a valid `str`, but I doubt that they form a valid `referenceLocator`. I strongly recommend refining its type further. This could be done by introducing a string subclass that does its own validation when instantiated.